### DISCUSSION TITLE
Primary Extended Spoken Language Code

### DISCUSSION THREAD

**Discussion by taussoe** [bbc/bmx#99] [Q&A] [answered]
Hi,

I’ve received an MXF sample file where certain metadata fields are set. Using ExifTool, I was able to find the following fields in the file:

Primary Extended Spoken Language Code (values: "sweL" and "sweR")
Framework Extended Text Language Code (values: "sweL" and "sweR")
These fields are located under [Track2]. Now, I would like to set these same fields in other MXF files using BMXLib. Is it possible to do this with BMXLib, specifically using bmxtranswrap or another tool in the library? If so, could you provide some guidance on how to achieve it?

Thanks in advance for your help!

**Comment by philipnbbc** [comment, marked as answer]
These DMS-1 (SMPTE 380M-2004 Descriptive Metadata Scheme 1) properties are not supported by bmx. bmx supports the more recent SMPTE ST 377-4:2021/2012 Multichannel Audio Labeling Framework which uses a different means to assign audio labels in MXF files.

For reference, I found these [MXF tag names](https://exiftool.org/TagNames/MXF.html) in ExifTool (e.g. PrimaryExtendedSpokenLanguageCode), traced to the [source code](https://github.com/exiftool/exiftool/blob/master/lib/Image/ExifTool/MXF.pm) (e.g. [PrimaryExtendedSpokenLanguageCode](https://github.com/exiftool/exiftool/blob/master/lib/Image/ExifTool/MXF.pm#L1954)) and the corresponding elements in the [SMPTE elements registry](https://registry.smpte-ra.org/view/published/Elements.xml) (e.g search for PrimaryExtendedSpokenLanguageCode or 060e2b34.01010107.03010102.03110000).

**Reply by taussoe** [reply]
Thank you so much for this info and for taking time to look into that specific tag. It's very much appreciated! It's an MXF metadata jungle out there :)
